pulpcore/cli/rpm/content.py

```diff
@@ -285,13 +323,152 @@ def upload(
     chunk_size: int,
     **kwargs: t.Any,
 ) -> None:
-    """Create a content unit by uploading a file"""
-    body: t.Dict[str, t.Any]
+    """Create a content unit by uploading a file or files"""
     if isinstance(entity_ctx, PulpRpmPackageContext):
-        result = entity_ctx.upload(file=file, chunk_size=chunk_size, **kwargs)
+        directory = kwargs.get("directory")
+        # Sanity: one of file|directory required
+        if (not (file or directory)) or (file and directory):
+            raise click.ClickException(
+                _("You must specify one (and only one) of --file or --directory.")
+            )
+
+        # Sanity: If directory, repository required
+        final_dest_repo_ctx = kwargs["repository"]
+        if directory and not final_dest_repo_ctx:
+            raise click.ClickException(
+                _("You must specify a --repository to use --directory uploads.")
+            )
+
+        # Sanity: ignore publish|use_temp unless directory has been specified
+        use_tmp = final_dest_repo_ctx and kwargs["use_temp_repository"]
+        do_publish = final_dest_repo_ctx and kwargs["publish"]
+
+        # Sanity: ignore relative_path if directory specified
+        if directory and kwargs["relative_path"]:
+            print(_("relative_path ignored on --directory uploads."))
+
+        if file:
+            # Single-file upload
+            result = entity_ctx.upload(
+                file=file,
+                chunk_size=chunk_size,
+                repository=final_dest_repo_ctx,
+                relative_path=kwargs["relative_path"],
+            )
+        else:
+            # Upload a directory-full of RPMs
+            try:
+                dest_repo_ctx = _determine_upload_repository(final_dest_repo_ctx, pulp_ctx, use_tmp)
+                result = _upload_rpms(entity_ctx, dest_repo_ctx, directory, chunk_size)
+                if use_tmp:
+                    result = _copy_to_final(dest_repo_ctx, final_dest_repo_ctx, pulp_ctx)
+            finally:
+                if use_tmp and dest_repo_ctx:
+                    # Remove the tmp-upload repo
+                    dest_repo_ctx.delete()
+
+        final_version_number = _latest_version_number(final_dest_repo_ctx)
+        if final_version_number:
+            print(
```

Comment:
Again. Better use `click.echo(..., err=True)`, because the output on stdout should match the `--format` option.